### PR TITLE
maximum and minmum qp setting for HEVC encoder.

### PR DESCRIPTION
**Pull request by hongbolv**

### CODE REVIEW

**Comment by dmitryermilov** [comment]
@thomasli21801 , please review

File: _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_legacy.cpp
Comment (thomasli21801) [inline _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_legacy.cpp:2406]: Hi Hongbo,

Since this max/min QP is per sequence settings, do we still need P/B settings here? It may not work, right?

Thanks,
Thomas

File: _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_data.h
Comment (thomasli21801) [inline _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_data.h:786]: Hi Hongbo,

May I know if any check for minQP/maxQP in valid range?

Thanks,
Thomas

Comment (hongbolv) [reply _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_data.h:786]: The range is (1, 51).

File: _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_legacy.cpp
Comment (hongbolv) [reply _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_legacy.cpp:2406]: The sequence setting is enough.

File: _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_data.h
Comment (dmitryermilov) [reply _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_data.h:786]: It should be higher than 51 for 10bit.

Comment (hongbolv) [reply _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_data.h:786]: The mediasdk library will add the offset so that it is higher than 51.

Comment (thomasli21801) [reply _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_data.h:786]: I mean if there is any implementation to check minQP/maxQP is within this range?

Comment (hongbolv) [reply _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_data.h:786]: Updated with range check in the VA layer.

Comment (thomasli21801) [reply _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_data.h:786]: Thanks Hongbo! The check looks good to me. But any chance to put to a common place for Windows and Linux? Any requirement on Windows?

Comment (hongbolv) [reply _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_data.h:786]: Thanks, Thomas! This is from Linux requirement. I have no idea about the Windows requirement.

Comment (thomasli21801) [reply _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_data.h:786]: that's OK. let me figure out Windows changes later.

**Review by thomasli21801 — DISMISSED** [review]

**Comment by dmitryermilov** [comment]
@hongbolv , can you please fix Windows CI build?

**Comment by hongbolv** [comment]
> @hongbolv , can you please fix Windows CI build?

@dmitryermilov I can't access the fail detail. Can you help check what's build error?

**Comment by dmitryermilov** [comment]
> > @hongbolv , can you please fix Windows CI build?
> 
> @dmitryermilov I can't access the fail detail. Can you help check what's build error?

Sorry, Hongbo. Its seems to be our internal infrastructure issue. Checking with maintainers..

**Comment by dmitryermilov** [comment]
@hongbolv , most likely windows CI failure is due to https://github.com/Intel-Media-SDK/MediaSDK/blob/c9809b1412ea25add743e288b9942ebf5591fab0/samples/sample_multi_transcode/src/transcode_utils.cpp#L2694 . Please move parsing of new cmd params to ParseAdditionalParams()

**Comment by dvrogozh** [comment]
Please, rebase the change to fetch CI fixes for clang-13, please.

**Comment by hongbolv** [comment]
> @hongbolv , most likely windows CI failure is due to
> 
> https://github.com/Intel-Media-SDK/MediaSDK/blob/c9809b1412ea25add743e288b9942ebf5591fab0/samples/sample_multi_transcode/src/transcode_utils.cpp#L2694
> 
> . Please move parsing of new cmd params to ParseAdditionalParams()

Done

File: _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_data.h
Comment (dmitryermilov) [reply _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_data.h:786]: @thomasli21801 , please approve if it looks good to you.

**Review by dmitryermilov — DISMISSED** [review]

**Review by thomasli21801 — DISMISSED** [review]

**Comment by dmitryermilov** [comment]
@hongbolv ,

can you please fix windows issue below?
```
c:\bb\b-1-27\mediasdk-windows\build_dir\repos\mediasdk\samples\sample_multi_transcode\src\pipeline_transcode.cpp(2574): error C2220: warning treated as error - no 'object' file generated [C:\bb\b-1-27\mediasdk-windows\build_dir\repos\MediaSDK\samples\sample_multi_transcode\sample_multi_transcode.vcxproj]
c:\bb\b-1-27\mediasdk-windows\build_dir\repos\mediasdk\samples\sample_multi_transcode\src\pipeline_transcode.cpp(2574): warning C4244: '=': conversion from 'mfxU16' to 'mfxU8', possible loss of data [C:\bb\b-1-27\mediasdk-windows\build_dir\repos\MediaSDK\samples\sample_multi_transcode\sample_multi_transcode.vcxproj]
c:\bb\b-1-27\mediasdk-windows\build_dir\repos\mediasdk\samples\sample_multi_transcode\src\pipeline_transcode.cpp(2575): warning C4244: '=': conversion from 'mfxU16' to 'mfxU8', possible loss of data [C:\bb\b-1-27\mediasdk-windows\build_dir\repos\MediaSDK\samples\sample_multi_transcode\sample_multi_transcode.vcxproj]
```

**Review by dmitryermilov — APPROVED** [review]

**Pull request merged**
